components/mpas-ocean/src/shared/mpas_ocn_tracer_surface_restoring.F

```diff
@@ -250,7 +251,11 @@ subroutine ocn_get_surfaceSalinityData( streamManager, &
         integer :: ierr
 
         integer, parameter :: nSums = 2
-        real (kind=RKIND), dimension(nSums) :: reductions, sums
+        real (kind=RKIND), dimension(nSums) :: reductions
+
+        integer, dimension(nSums) :: indexForReproSum
```

Comment:
@philipwjones It's a minor detail, but I'm curious. Can you confirm if I understood this correctly: in this case, it's a coincidence that indexForReproSum and reductions have the same dimensions of nSums? 
Technically, indexForReproSum should always have 2 (the min, max you mentioned), while nSums could be anything given the needs of the routine? (here, 2 since we need the area sum and the area-weighted salinity sum.
Does it make sense to use nSums to define indexForReproSum then?

Reply:
@alicebarthel  Good catch. The index array should be dimensioned 2*(number of dimensions being summed over) - in this case just 2 since the arrays are just 1-d. So nSums is coincidentally correct but in incorrect in general.

Reply:
@alicebarthel I wouldn't say it is "a coincidence" that nSums=2;  this routine isn't meant to be general which is why I hardwired it to be correct for this case.

Reply:
ok, maybe I still don't understand the question.  we have 2 sums to do:  area and area*deltaS, so nSums = 2.  so we need the same number of outputs from the 2 global sum calls:  global_sum(area) and global_sum(area*deltaS) which are put into reductions(1) and reductions(2), respectively.  so reductions should always be dimensioned nSums.  I guess I could have just put sumAreaGlobal and sumAreaDeltaSGlobal directly in the global sum calls and not needed reductions(), but I wanted to keep the new version as similar to the current version as possible.  does this make sense?

Reply:
I don't have an issue with dimensioning reductions with nSums, I think that's great! 
Dimensioning indexForReproSum with nSums (L256) makes no sense to me, since the length (2) has nothing to do with nSums, but with 2*len(shape(variable we are summing over)) = 2*1-D variable = 2 in this case. My comment is over L256, not 254, sorry if the github visualization is confusing.

Reply:
ah, ok, now I understand.  you're right, that is misleading.  should I just place it with "2"?

Reply:
and a comment?  thanks for clarifying.

Reply:
I think you should create a variable ThroatWarblerMangrove that's pronounced "two"...

Reply:
@maltrud Sounds good to me, thanks!